jwst/flatfield/flat_field_step.py

```diff
@@ -87,10 +90,15 @@ def process(self, input):
                 'user_supplied_flat': datamodels.open(self.user_supplied_flat)
             }
 
+            # Record the user-supplied flat as the FLAT reference type for recording
+            # in the result header.
+            self._reference_files_used.append(('flat', Path(self.user_supplied_flat).name))
```

Comment:
What happens here if the user supplied flat is just in the form of a datamodel, as opposed to an actual file on disk? And isn't that the way it's going to work in the nrsmsb stuff in calwebb_spec2, i.e. we'll just be passing in a previously computed datamodel?

Reply:
Right, this would fail painfully. 👍

Reply:
Anything worth failing is worth failing painfully.